src/main/groovy/com/spidasoftware/schema/conversion/changeset/v12/CSAMaxWindTemperatureChangeset.groovy

```diff
@@ -0,0 +1,106 @@
+            changed |= applyTemperatureToLoadCase(analysisCase)
+        }
+        return changed
+    }
+
+    @Override
+    void applyToProject(Map projectJSON) throws ConversionException {
+        super.applyToProject(projectJSON)
+        projectJSON.defaultLoadCases?.each { Map loadCase ->
+            applyTemperatureToLoadCase(loadCase)
+        }
+    }
+
+    @Override
+    void applyToDesign(Map designJSON) throws ConversionException {
+        super.applyToDesign(designJSON)
+        designJSON.analysis?.each { Map analysis ->
+            if (analysis.analysisCaseDetails != null) {
+                applyTemperatureToLoadCase(analysis.analysisCaseDetails as Map)
+            }
+        }
+    }
+
+    @Override
+    boolean applyToResults(Map resultsJSON) throws ConversionException {
+        boolean changed = super.applyToResults(resultsJSON)
+        resultsJSON.results?.each { Map result ->
+            if (result.analysisCaseDetails != null) {
+                changed |= applyTemperatureToLoadCase(result.analysisCaseDetails as Map)
+            }
+        }
+        return changed
+    }
+
+    @Override
+    boolean revertClientData(Map clientDataJSON) throws ConversionException {
+        boolean changed = false
+        clientDataJSON.analysisCases?.each { Map analysisCase ->
+            changed |= revertTemperatureFromLoadCase(analysisCase)
+        }
+        return changed
+    }
+
+    @Override
+    void revertProject(Map projectJSON) throws ConversionException {
+        super.revertProject(projectJSON)
+        projectJSON.defaultLoadCases?.each { Map loadCase ->
+            revertTemperatureFromLoadCase(loadCase)
+        }
+    }
+
+    @Override
+    void revertDesign(Map designJSON) throws ConversionException {
+        super.revertDesign(designJSON)
+        designJSON.analysis?.each { Map analysis ->
+            if (analysis.analysisCaseDetails != null) {
+                revertTemperatureFromLoadCase(analysis.analysisCaseDetails as Map)
+            }
+        }
+    }
+
+    @Override
+    boolean revertResults(Map resultsJSON) throws ConversionException {
+        boolean changed = super.revertResults(resultsJSON)
+        resultsJSON.results?.each { Map result ->
+            if (result.analysisCaseDetails != null) {
+                changed |= revertTemperatureFromLoadCase(result.analysisCaseDetails as Map)
+            }
+        }
+        return changed
+    }
+
+    boolean applyTemperatureToLoadCase(Map loadCaseJSON) {
+        if (loadCaseJSON.type == "CSA 2020 Maximum Wind") {
+            (loadCaseJSON.overrides as Map).temperature = [value: 15.0, unit: "CELSIUS"]
+            (loadCaseJSON.valuesApplied as Map).temperature = [value: 15.0, unit: "CELSIUS"]
+            return true
+        }
+        return false
+    }
+
+    boolean revertTemperatureFromLoadCase(Map loadCaseJSON) {
+        return (loadCaseJSON.overrides as Map)?.remove("temperature") != null
```

Comment:
we should not do this if the temperature override is anything other than 15 C

Reply:
This is because the user could have manually set an override themselves? What if the user manually set it to 15 C in the old schema version? In that case, we would not create a temperature override (because a value is already present) and then we would remove an override that the user wanted